xarray/core/rolling.py

```diff
@@ -0,0 +1,164 @@
+import numpy as np
+
+from .pycompat import OrderedDict, zip
+from .common import ImplementsRollingArrayReduce, empty_like
+from .combine import concat
+from .ops import inject_bottleneck_rolling_methods
+
+
+class Rolling(object):
+    """A object that implements the moving window pattern.
+
+    See Also
+    --------
+    Dataset.groupby
+    DataArray.groupby
+    Dataset.rolling
+    DataArray.rolling
+    """
+
+    _attributes = ['window', 'min_periods', 'center', 'dim']
+
+    def __init__(self, obj, min_periods=None, center=False, **kwarg):
+        """
+        Moving window object.
+
+        Parameters
+        ----------
+        obj : Dataset or DataArray
+            Object to window.
+        min_periods : int, default None
+            Minimum number of observations in window required to have a value
+            (otherwise result is NA). The default, None, is equivalent to
+            setting min_periods equal to the size of the window.
+        center : boolean, default False
+            Set the labels at the center of the window.
+        kwarg : dim=window
```

Comment:
Use `**kwarg` instead. Also, I would consider calling this argument `windows` instead.